lib/rails/pagination.rb

```diff
@@ -29,8 +29,8 @@ def _paginate_collection(collection, options={})
 
       collection = ApiPagination.paginate(collection, options)
 
-      links = (headers['Link'] || "").split(',').map(&:strip)
-      url   = request.original_url.sub(/\?.*$/, '')
+      links = (headers['Link'] || '').split(',').map(&:strip)
+      url   = base_url + request.original_fullpath.sub(/\?.*$/, '')
```

Comment:
While we're already here, could you just change this to `request.path_info` so we can get rid of the unnecessary `sub`?